usl_pipeline/usl_lib/usl_lib/transformers/feature_raster_transformers.py

```diff
@@ -127,3 +129,55 @@ def transform_to_feature_raster_layers(
             infiltration_raster_layers,
         )
     )
+
+
+def rescale_feature_blob(
+    feature_matrix_input_fd: typing.BinaryIO,
+    study_area_metadata: metastore.StudyArea,
+    feature_matrix_output_fd: typing.BinaryIO,
+) -> None:
+    """Performs scaling of elevation data in the 0-th layer of feature matrix.
+
+    Args:
+        feature_matrix_input_fd: Input stream to read unscaled feature matrix from.
+        study_area_metadata: Study Area metadata containing min/max elevation values
+            that are used for scaling.
+        feature_matrix_output_fd: Output stream to write scaled results to.
+    """
+    unscaled_feature_matrix = numpy.load(feature_matrix_input_fd)
+    scaled_feature_matrix = rescale_feature_matrix(
+        unscaled_feature_matrix, study_area_metadata
+    )
+    numpy.save(feature_matrix_output_fd, scaled_feature_matrix)
+
+
+def rescale_feature_matrix(
+    unscaled_feature_matrix: npt.NDArray,
+    study_area_metadata: metastore.StudyArea,
+) -> npt.NDArray:
+    """Performs scaling of elevation data in the 0-th layer of feature matrix.
+
+    Args:
+        unscaled_feature_matrix: Unscaled feature matrix to perform scaling for.
+        study_area_metadata: Study Area metadata containing min/max elevation values
+            that are used for scaling.
+
+    Returns:
+        Scaled feature matrix.
+    """
+    shape = unscaled_feature_matrix.shape
+    feature_layers = [
+        layer.reshape(shape[0], shape[1])
+        for layer in numpy.dsplit(unscaled_feature_matrix, shape[2])
```

Comment:
We don't need to dsplit the arrays then dstack them back together. We can select along the third dimension directly. So we can do
```python
elevation_data = unscaled_feature_matrix[:, :, 0]
presence_mask = unscaled_feature_matrix[:, :, 1]
elevation_data[:] = elevation_data[presence_max == 1] = 
        elevation_data[presence_mask == 1] - elevation_min
    ) / (elevation_max - elevation_min)
```
which would mutate `unscaled_feature_matrix` in place.

Reply:
Done